mysql/mysqldump.go

```diff
@@ -0,0 +1,578 @@
+			}
+			// Likely causes of failing to parse:
+			// a) complex statements with embedded semicolons e.g. 'CREATE FUNCTION'
+			// b) a semicolon embedded in a multi-line comment, or
+			// c) a semicolon embedded a string constant or column/table name.
+			// We deal with this case by reading another line and trying again.
+			conv.Stats.Reparsed++
+		}
+		if r.EOF {
+			return nil, nil, fmt.Errorf("Error parsing last %d line(s) of input", len(l))
+		}
+	}
+}
+
+// processStatement extracts schema information from MYSQL
+// statements, updating Conv with new schema information, and returning
+// true if INSERT statement is encountered.
+func processStatement(conv *internal.Conv, stmt ast.StmtNode) bool {
+
+	switch ddlStmt := stmt.(type) {
+	case *ast.CreateTableStmt:
+		if conv.SchemaMode() {
+			processCreateTable(conv, ddlStmt)
+		}
+	case *ast.AlterTableStmt:
+		if conv.SchemaMode() {
+			processAlterTable(conv, ddlStmt)
+		}
+	case *ast.SetStmt:
+		if conv.SchemaMode() {
+			processSetStmt(conv, ddlStmt)
+		}
+	case *ast.InsertStmt:
+		return true
+	default:
+		conv.SkipStatement(reflect.TypeOf(stmt).String())
+	}
+	return false
+}
+
+func processSetStmt(conv *internal.Conv, stmt *ast.SetStmt) {
+	if stmt.Variables != nil && len(stmt.Variables) > 0 {
+		for _, variable := range stmt.Variables {
+			if variable.Name == "TIME_ZONE" {
+				value := variable.Value
+				switch val := value.(type) {
+				case *driver.ValueExpr:
+					if val.GetValue() == nil {
+						logStmtError(conv, stmt, fmt.Errorf("found nil value in set statement"))
+						return
+					}
+					conv.TimezoneOffset = fmt.Sprintf("%v", val.GetValue())
+				default:
+					logStmtError(conv, stmt, fmt.Errorf("found %s type value in set statement", reflect.TypeOf(val)))
+					return
+				}
+			}
+		}
+	}
+}
+
+func processCreateTable(conv *internal.Conv, stmt *ast.CreateTableStmt) {
+	var colNames []string
+	colDef := make(map[string]schema.Column)
+	if stmt.Table == nil {
+		logStmtError(conv, stmt, fmt.Errorf("table is nil"))
+		return
+	}
+	tableName, err := getTableName(stmt.Table)
+	if err != nil {
+		logStmtError(conv, stmt, fmt.Errorf("can't get table name: %w", err))
+		return
+	}
+	columns := stmt.Cols
+	var keys []schema.Key
+	for _, element := range columns {
+		colname, col, isPk, err := processColumn(conv, tableName, element)
+		if err != nil {
+			logStmtError(conv, stmt, err)
+			return
+		}
+		colNames = append(colNames, colname)
+		colDef[colname] = col
+		if isPk {
+			keys = append(keys, schema.Key{Column: colname})
+		}
+	}
+	conv.SchemaStatement(reflect.TypeOf(stmt).String())
+	conv.SrcSchema[tableName] = schema.Table{
+		Name:        tableName,
+		ColNames:    colNames,
+		ColDefs:     colDef,
+		PrimaryKeys: keys}
+	for _, constraint := range stmt.Constraints {
+		processConstraint(conv, tableName, constraint, "CREATE TABLE")
+	}
+}
+
+func processConstraint(conv *internal.Conv, table string, constraint *ast.Constraint, s string) {
```

Comment:
I like short var names, but in general func params should be a bit more descriptive.

Suggest something like stmtType?

Same comment for checkEmpty.

Reply:
Yup, I have tried to follow pgdump.go as much as possible. 
No issues. I will change it to `stmtType` in mysqldump.go and pgdump.go as well.